state-chain/pallets/cf-ingress-egress/src/tests.rs

```diff
@@ -1576,3 +1566,113 @@ fn can_update_multiple_items_at_once() {
 		assert_eq!(MinimumDeposit::<Test, _>::get(eth::Asset::Eth), 200);
 	});
 }
+
+#[test]
+fn handle_prewitness_deposit() {
+	const ASSET: cf_chains::assets::eth::Asset = eth::Asset::Eth;
+	const DEPOSIT_AMOUNT: u128 = 50000;
+
+	new_test_ext().execute_with(|| {
+		let (_id, address, ..) =
+			IngressEgress::request_liquidity_deposit_address(ALICE, ASSET, 0).unwrap();
+		let deposit_address: <Ethereum as Chain>::ChainAccount = address.try_into().unwrap();
+
+		let deposit_detail: DepositWitness<Ethereum> = DepositWitness::<Ethereum> {
+			deposit_address,
+			asset: ASSET,
+			amount: DEPOSIT_AMOUNT,
+			deposit_details: (),
+		};
+
+		let deposit_witnesses = vec![deposit_detail.clone()];
+
+		// Submit the prewitnessed deposit
+		assert_ok!(IngressEgress::add_prewitnessed_deposits(deposit_witnesses.clone(),));
+
+		// Check that the deposit is stored in the storage
+		let prewitnessed_deposit_id = PrewitnessedDepositIdCounter::<Test>::get();
+		let channel_id = ChannelIdCounter::<Test>::get();
+		assert_eq!(
+			PrewitnessedDeposits::<Test>::get(channel_id, prewitnessed_deposit_id),
+			Some(PrewitnessedDeposit { asset: ASSET, amount: DEPOSIT_AMOUNT, deposit_address })
+		);
+	});
+}
+
+#[test]
+fn should_cleanup_prewitnessed_deposits_when_channel_is_recycled() {
+	const ASSET: cf_chains::assets::eth::Asset = eth::Asset::Eth;
+	const DEPOSIT_AMOUNT: u128 = 50000;
+
+	new_test_ext().execute_with(|| {
+		// Create a deposit channel
+		let (_id, address, ..) =
+			IngressEgress::request_liquidity_deposit_address(ALICE, ASSET, 0).unwrap();
+		let deposit_address: <Ethereum as Chain>::ChainAccount = address.try_into().unwrap();
+
+		// Submit the prewitnessed deposit
+		let deposit_detail: DepositWitness<Ethereum> = DepositWitness::<Ethereum> {
+			deposit_address,
+			asset: ASSET,
+			amount: DEPOSIT_AMOUNT,
+			deposit_details: (),
+		};
+		let deposit_witnesses = vec![deposit_detail.clone()];
+		assert_ok!(IngressEgress::add_prewitnessed_deposits(deposit_witnesses.clone(),));
+
+		// Check that the deposit is stored in the storage
+		let prewitnessed_deposit_id = PrewitnessedDepositIdCounter::<Test>::get();
+		let channel_id = ChannelIdCounter::<Test>::get();
+		assert_eq!(
+			PrewitnessedDeposits::<Test>::get(channel_id, prewitnessed_deposit_id),
+			Some(PrewitnessedDeposit { asset: ASSET, amount: DEPOSIT_AMOUNT, deposit_address })
+		);
+
+		// Fast forward the block height to the recycle block of the created deposit channel
+		let expiry_block = IngressEgress::expiry_and_recycle_block_height().2;
+		BlockHeightProvider::<MockEthereum>::set_block_height(expiry_block);
+
+		// Run the cleanup
+		IngressEgress::on_idle(1, Weight::MAX);
```

Comment:
Looks like `1` isn't used in `on_idle`, but it is meant to be block height, which you are setting to `expiry_block`, so I think it would be more correct to use `expiry_block` here too.

Reply:
`expiry_block` is the eth block, but the `1` is the SC block. So it does not need to be the same value. I will set it to default to show that it does not matter for the test.